numba/core/typing/new_builtins.py

```diff
@@ -0,0 +1,1178 @@
+
+    cases += [signature(op, op) for op in sorted(types.py_real_domain)]
+    cases += [signature(op, op) for op in sorted(types.np_real_domain)]
+
+    cases += [signature(op, op) for op in sorted(types.py_complex_domain)]
+    cases += [signature(op, op) for op in sorted(types.np_complex_domain)]
+
+    cases += [signature(types.py_intp, types.py_bool)]
+    cases += [signature(types.np_intp, types.np_bool_)]
+
+
+@infer_global(operator.neg)
+class UnaryNegate(UnaryOp):
+    pass
+
+
+@infer_global(operator.pos)
+class UnaryPositive(UnaryOp):
+   pass
+
+
+@infer_global(operator.not_)
+class UnaryNot(ConcreteTemplate):
+    cases = [signature(types.np_bool_, types.np_bool_)]
+    cases += [signature(types.np_bool_, op) for op in sorted(types.np_signed_domain)]
+    cases += [signature(types.np_bool_, op) for op in sorted(types.np_unsigned_domain)]
+    cases += [signature(types.np_bool_, op) for op in sorted(types.np_real_domain)]
+    cases += [signature(types.np_bool_, op) for op in sorted(types.np_complex_domain)]
+
+
+class OrderedCmpOp(ConcreteTemplate):
+    cases = [signature(types.np_bool_, types.np_bool_, types.np_bool_)]
+    cases += [signature(types.np_bool_, op, op) for op in sorted(types.np_signed_domain)]
+    cases += [signature(types.np_bool_, op, op) for op in sorted(types.np_unsigned_domain)]
+    cases += [signature(types.np_bool_, op, op) for op in sorted(types.np_real_domain)]
+
+
+class UnorderedCmpOp(ConcreteTemplate):
+    cases = OrderedCmpOp.cases + [
+        signature(types.np_bool_, op, op) for op in sorted(types.np_complex_domain)]
+
+
+@infer_global(operator.lt)
+class CmpOpLt(OrderedCmpOp):
+    pass
+
+
+@infer_global(operator.le)
+class CmpOpLe(OrderedCmpOp):
+    pass
+
+
+@infer_global(operator.gt)
+class CmpOpGt(OrderedCmpOp):
+    pass
+
+
+@infer_global(operator.ge)
+class CmpOpGe(OrderedCmpOp):
+    pass
+
+
+# more specific overloads should be registered first
+@infer_global(operator.eq)
+class ConstOpEq(AbstractTemplate):
+    def generic(self, args, kws):
+        assert not kws
+        (arg1, arg2) = args
+        if isinstance(arg1, types.Literal) and isinstance(arg2, types.Literal):
+            return signature(types.np_bool_, arg1, arg2)
+
+
+@infer_global(operator.ne)
+class ConstOpNotEq(ConstOpEq):
+    pass
+
+
+@infer_global(operator.eq)
+class CmpOpEq(UnorderedCmpOp):
+    pass
+
+
+@infer_global(operator.ne)
+class CmpOpNe(UnorderedCmpOp):
+    pass
+
+
+class TupleCompare(AbstractTemplate):
+    def generic(self, args, kws):
+        [lhs, rhs] = args
+        if isinstance(lhs, types.BaseTuple) and isinstance(rhs, types.BaseTuple):
+            for u, v in zip(lhs, rhs):
+                # Check element-wise comparability
+                res = self.context.resolve_function_type(self.key, (u, v), {})
+                if res is None:
+                    break
+            else:
+                return signature(types.np_bool_, lhs, rhs)
```

Comment:
```suggestion
                return signature(types.py_bool, lhs, rhs)
```
this is a Python builtin so should remain in the Python type domain?